prow/clonerefs/run.go

```diff
@@ -73,3 +87,41 @@ func (o Options) Run() error {
 
 	return nil
 }
+
+// addSSHKeys will start the ssh-agent and add all the specified
+// keys, returning the ssh-agent environment variables for reuse
+func addSSHKeys(paths []string) ([]string, error) {
+	vars, err := exec.Command("ssh-agent").CombinedOutput()
+	if err != nil {
+		return []string{}, fmt.Errorf("failed to start ssh-agent: %v", err)
+	}
+	logrus.Info("Started SSH agent")
+	// ssh-agent will output three lines of text, in the form:
+	// SSH_AUTH_SOCK=xxx; export SSH_AUTH_SOCK;
+	// SSH_AGENT_PID=xxx; export SSH_AGENT_PID;
+	// echo Agent pid xxx;
+	// We need to parse out the environment variables from that.
+	parts := strings.Split(string(vars), ";")
+	env := []string{strings.TrimSpace(parts[0]), strings.TrimSpace(parts[2])}
+	for _, keyPath := range paths {
+		// we can be given literal paths to keys or paths to dirs
+		// that are mounted from a secret, so we need to check which
+		// we have
+		if err := filepath.Walk(keyPath, func(path string, info os.FileInfo, err error) error {
+			if info == nil || info.IsDir() || strings.HasPrefix(info.Name(), "..") {
```

Comment:
We need to skip directories starting with ".." instead of walking through them to avoid errors like:
```
{"component":"clonerefs","error":"failed to add SSH keys: failed to add ssh key at /secrets/ssh/ssh-secret/..2018_05_11_03_21_20.325268827/ssh-secret: exit status 1","level":"fatal","msg":"Failed to clone refs","time":"2018-05-11T03:21:28Z"}
```